app/models/user.rb

```diff
@@ -16,9 +18,9 @@ class User < ActiveRecord::Base
   attr_accessible *(ACCESSIBLE_ATTRIBUTES + [:admin]), :as => :admin
 
   validates_presence_of :username
-  validates_uniqueness_of :username
+  validates :username, uniqueness: { case_sensitive: false }
```

Comment:
The default collation for a utf8 varchar/char column is either utf8_general_ci (MySQL) or utf8_unicode_ci (ActiveRecord), and they are both case-insensitive.